src/network/networkd-link.c

```diff
@@ -743,6 +744,9 @@ void link_check_ready(Link *link) {
         if (!link->addresses_configured)
                 return;
 
+        if (!link->neighbors_configured)
+                return;
```

Comment:
we should consider this long time issue too . https://github.com/systemd/systemd/issues/9077